src/core/layout/qgslayoutaligner.cpp

```diff
@@ -97,6 +99,65 @@ void QgsLayoutAligner::distributeItems( QgsLayout *layout, const QList<QgsLayout
   if ( items.size() < 2 )
     return;
 
+  double minCoord = std::numeric_limits<double>::max();
+  double maxCoord = std::numeric_limits<double>::lowest();
+  QMap< double, QgsLayoutItem * > itemCoords;
+
+  double step;
+
+  // equispaced distribution doesn't follow the same approach of the other distribution types
+  if ( distribution == DistributeHSpace || distribution == DistributeVSpace )
```

Comment:
Can you shuffle this code out into a new function to avoid the distributeItems function become too long?